Source/WebCore/dom/Range.cpp

```diff
@@ -317,95 +319,111 @@ ExceptionOr<RefPtr<DocumentFragment>> Range::processContents(ActionType action)
         return fragment;
     }
 
-    // Since mutation events can modify the range during the process, the boundary points need to be saved.
-    RangeBoundaryPoint originalStart(m_start);
-    RangeBoundaryPoint originalEnd(m_end);
-
-    // what is the highest node that partially selects the start / end of the range?
-    RefPtr<Node> partialStart = highestAncestorUnderCommonRoot(&originalStart.container(), commonRoot.get());
-    RefPtr<Node> partialEnd = highestAncestorUnderCommonRoot(&originalEnd.container(), commonRoot.get());
-
-    // Start and end containers are different.
-    // There are three possibilities here:
-    // 1. Start container == commonRoot (End container must be a descendant)
-    // 2. End container == commonRoot (Start container must be a descendant)
-    // 3. Neither is commonRoot, they are both descendants
-    //
-    // In case 3, we grab everything after the start (up until a direct child
-    // of commonRoot) into leftContents, and everything before the end (up until
-    // a direct child of commonRoot) into rightContents. Then we process all
-    // commonRoot children between leftContents and rightContents
-    //
-    // In case 1 or 2, we skip either processing of leftContents or rightContents,
-    // in which case the last lot of nodes either goes from the first or last
-    // child of commonRoot.
-    //
-    // These are deleted, cloned, or extracted (i.e. both) depending on action.
-
-    // Note that we are verifying that our common root hierarchy is still intact
-    // after any DOM mutation event, at various stages below. See webkit bug 60350.
-
-    RefPtr<Node> leftContents;
-    if (&originalStart.container() != commonRoot && commonRoot->contains(&originalStart.container())) {
-        auto firstResult = processContentsBetweenOffsets(action, nullptr, &originalStart.container(), originalStart.offset(), originalStart.container().length());
-        auto secondResult = processAncestorsAndTheirSiblings(action, &originalStart.container(), ProcessContentsForward, WTFMove(firstResult), commonRoot.get());
-        // FIXME: A bit peculiar that we silently ignore the exception here, but we do have at least some regression tests that rely on this behavior.
-        if (!secondResult.hasException())
-            leftContents = secondResult.releaseReturnValue();
-    }
+    Vector<Ref<Element>> elementsToUpgrade;
+    {
+        CustomElementReactionStack customElementsReactionHoldingTank(commonRoot->document().globalObject());
+
+        // Since mutation events can modify the range during the process, the boundary points need to be saved.
+        RangeBoundaryPoint originalStart(m_start);
+        RangeBoundaryPoint originalEnd(m_end);
+
+        // what is the highest node that partially selects the start / end of the range?
+        RefPtr<Node> partialStart = highestAncestorUnderCommonRoot(&originalStart.container(), commonRoot.get());
```

Comment:
Would `RefPtr` suffice instead of `RefPtr<Node>`?

Reply:
Sure.